test/test_population.py

```diff
@@ -76,3 +77,48 @@ def test_parent_individuals_are_assigned_correct_indices(population_params, geno
 
     for idx, ind in enumerate(pop.parents):
         assert ind.idx == idx
+
+
+def test_dna_init(population_params, genome_params):
+
+    dna = [
+        ID_INPUT_NODE,
+        ID_NON_CODING_GENE,
+        ID_NON_CODING_GENE,
+        0,
+        0,
+        0,
+        1,
+        1,
+        1,
+        2,
+        2,
+        2,
+        ID_OUTPUT_NODE,
+        3,
+        ID_NON_CODING_GENE,
+    ]
+
+    def dna_init(ind):
+        ind.genome.dna = dna
+        return ind
+
+    genome_params = {
+        "n_inputs": 1,
+        "n_outputs": 1,
+        "n_columns": 3,
+        "n_rows": 1,
+        "levels_back": None,
+        "primitives": (cgp.Add, cgp.Sub, cgp.ConstantFloat),
+    }
+
+    # without passing dna_init comparison fails
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    for ind in pop.parents:
+        with pytest.raises(AssertionError):
```

Comment:
Couldn't this fail by chance?

Reply:
it could indeed, it's just _very_ unlikely. i'm too lazy to do the correct computation but just assume three options for each gene, 15 genes, the probability of generating exactly this dna randomly would be something like ~10^-7, so not very likely ;)